src/backend/Repositories/OffersRepository.cs

```diff
@@ -53,5 +54,16 @@ public List<Offer> GetAll()
                 .ToList();
             return offers;
         }
+
+        public Offer UpdateOffer (Offer oldOffer, UpdateOfferDto newOffer)
+        {
+            _db.Entry(oldOffer).CurrentValues.SetValues(newOffer);
```

Comment:
This method overwrites old values with new values from the given object. That's why you cannot send a request with e. g. quantity not provided since it overrides the old quantity with an invalid value and fails the `[Required]` check.